iotdb/src/main/java/org/apache/iotdb/db/qp/logical/sys/PropertyOperator.java

```diff
@@ -56,6 +56,51 @@ public PropertyType getPropertyType() {
   }
 
   public enum PropertyType {
-    ADD_TREE, ADD_PROPERTY_LABEL, DELETE_PROPERTY_LABEL, ADD_PROPERTY_TO_METADATA, DEL_PROPERTY_FROM_METADATA
+    ADD_TREE, ADD_PROPERTY_LABEL, DELETE_PROPERTY_LABEL, ADD_PROPERTY_TO_METADATA, DEL_PROPERTY_FROM_METADATA;
```

Comment:
why not write like ADD_TREE（0）, ADD_PROPERTY_LABEL（1）....

Reply:
If implement like that, I think it will be hard to get enum value by index

Reply:
Each enum item should have a fixed value, in case of future changes.
p.s. In my opinion, it's also not a good practice to write switch-cases to convert between enum ordinal and name.

Reply:
> Each enum item should have a fixed value, in case of future changes.
> p.s. In my opinion, it's also not a good practice to write switch-cases to convert between enum ordinal and name.

Do you have any other elegant way to do the same function?

Reply:
> Each enum item should have a fixed value, in case of future changes.
> p.s. In my opinion, it's also not a good practice to write switch-cases to convert between enum ordinal and name.

I run a test to show that switch method is more efficient. In my test, I use both ordinal/values methods which you prefer and switch methods which I use, as the result shows, switch is more efficient than your method through writing more codes. So it's necessary to use the current implementation.

![image](https://user-images.githubusercontent.com/26211279/55204900-ace02780-520b-11e9-81da-a146b077c9f5.png)